FirebaseStorageUI/UIImageView+FirebaseStorage.m

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <objc/runtime.h>
+
+#import "UIImageView+FirebaseStorage.h"
+
+@interface UIImageView (FirebaseStorage_Private)
+@property (nonatomic, readwrite, nullable, setter=sd_setCurrentDownloadTask:) FIRStorageDownloadTask *sd_currentDownloadTask;
+@end
+
+@implementation UIImageView (FirebaseStorage)
+
+- (FIRStorageDownloadTask *)sd_setImageWithStorageReference:(FIRStorageReference *)storageRef {
+  return [self sd_setImageWithStorageReference:storageRef placeholderImage:nil completion:nil];
+}
+
+- (FIRStorageDownloadTask *)sd_setImageWithStorageReference:(FIRStorageReference *)storageRef
+                                           placeholderImage:(UIImage *)placeholder {
+  return [self sd_setImageWithStorageReference:storageRef placeholderImage:placeholder completion:nil];
+}
+
+- (FIRStorageDownloadTask *)sd_setImageWithStorageReference:(FIRStorageReference *)storageRef
+                                           placeholderImage:(UIImage *)placeholder
+                                                 completion:(void (^)(UIImage *_Nullable,
+                                                                      NSError *_Nullable,
+                                                                      SDImageCacheType,
+                                                                      FIRStorageReference *))completion {
+  return [self sd_setImageWithStorageReference:storageRef
+                                  maxImageSize:5e6 // 5 megabytes
```

Comment:
Make 10MB (some of the Apple system images are around this size). 20MB may even be more reasonable.